src/index.js

```diff
@@ -1,7 +1,9 @@
-import { CID } from './cid.js'
+import * as CID from './cid.js'
```

Comment:
I really dislike this because this shadows `CID` interface that otherwise would have been exported by line 7, which would have led to a lot more clear errors from the type checker in dependencies saying that `CID` is an interface not a class. But then I'm not sure how to export all of the `cid.js` functionality, I've considered `import * as cid from './cid.js'` but it's not ideal either because:

1. Most code would have to change (while now only type annotations have to change)
2. Code using this module usually create local variable named `cid`.

If you have a better ideal please let me know.

Reply:
After sleeping on this over night I think it may actually be better if we provided facade `CID` class that all the existing code would end up referring to and provided new `CID` interface and static functionality separately.

General idea is that facade CID will take interface CID in all the parameter positions and `CID` facade class in all the return positions. That way all existing code will start backwards compatible, which would allow gradual transition to interface.

On the flip side new code that will produce `CID` interface will be incompatible with old code that expects `CID` facade class.

In the end I think it's whether we want to stretch the pain over time or just take it all in one gulp.